### PR TITLE
chore: add config to support cjs

### PR DESCRIPTION
**Pull request by dankreiger**
### Issue

closes https://github.com/typicode/lowdb/issues/471

***

### Description 

This PR adds backwards compatibility to `lowdb` for `commonjs`.

### To Test

1. Reproduce error in a node project e.g. `express.js`
2. In this project, run the following command to install the version of this PR branch with access to project's `lib` directory.

   ```sh
   npm install github:dankreiger/lowdb#chore/esm-cjs-hybrid-WITH-LIB
   ```

2. Error should be fixed in https://github.com/typicode/lowdb/issues/471

***

### New Build Structure

```sh
lib/
├── cjs
│   ├── Low.js
│   ├── LowSync.js
│   ├── MissingAdapterError.js
│   ├── adapters
│   │   ├── JSONFile.js
│   │   ├── JSONFileSync.js
│   │   ├── LocalStorage.js
│   │   ├── Memory.js
│   │   ├── MemorySync.js
│   │   ├── TextFile.js
│   │   └── TextFileSync.js
│   ├── index.js
│   └── package.json
├── esm
│   ├── Low.js
│   ├── LowSync.js
│   ├── MissingAdapterError.js
│   ├── adapters
│   │   ├── JSONFile.js
│   │   ├── JSONFileSync.js
│   │   ├── LocalStorage.js
│   │   ├── Memory.js
│   │   ├── MemorySync.js
│   │   ├── TextFile.js
│   │   └── TextFileSync.js
│   ├── index.js
│   └── package.json
└── types
    ├── Low.d.ts
    ├── LowSync.d.ts
    ├── MissingAdapterError.d.ts
    ├── adapters
    │   ├── JSONFile.d.ts
    │   ├── JSONFileSync.d.ts
    │   ├── LocalStorage.d.ts
    │   ├── Memory.d.ts
    │   ├── MemorySync.d.ts
    │   ├── TextFile.d.ts
    │   └── TextFileSync.d.ts
    └── index.d.ts
 ```

### Note

- Sometimes I choose to do this with `rollup.js` or `webpack`, but I noticed this project is trying to stay slim, so I just stuck with the `typescript` build process.
- The separate builds inherit their respective module format by referencing the specified `type` in the nearest parent/sibling `package.json`. 
    - Source: https://nodejs.org/api/packages.html#packages_type.

### CODE REVIEW

File: scripts/hybrid.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (dankreiger) [inline scripts/hybrid.sh:1]: https://nodejs.org/api/packages.html#packages_type

File: scripts/hybrid.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# $1 = folder name
+# $2 = package type
+function create_package_json() {
+cat > lib/$1/package.json << EOF
+{
+  "type": "$2"
```
Comment (dankreiger) [inline scripts/hybrid.sh:8]: https://nodejs.org/api/packages.html#packages_type

**Review by hsinlive — APPROVED** [review]

**Comment by ckcr4lyf** [comment]
This works great. I cannot use lowdbv2 currently in my master branch since without this MR, it is a breaking change

**Comment by dankreiger** [comment]
@typicode is this an update you want? I can resolve the merge conflicts if so. 

I'm reading [elsewhere](https://github.com/typicode/lowdb/issues/462#issuecomment-851533956) that you envision this project build as a pure ESM module.

Thank you by the way for your wonderful open source projects! They have been so helpful to me over the years.

**Comment by typicode** [comment]
Hmm, I think I'll stay on pure ESM. Tools will probably catch up with ESM and other projects are going pure ESM as well.

In any case, thanks a lot for the PR! :+1: 

For anyone having trouble with lowdb 2, here's a very good guide regarding how to use/migrate to ESM:
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#how-can-i-move-my-commonjs-project-to-esm

If you can't migrate to ESM right now, I'd suggest staying on lowdb v1 for now.

**Comment by mouafa** [comment]
@typicode what's the problem if both ESM and CJS are published, you still write the code the same way, it'll just give users and engines the possibility to pick whatever is more convenient. assuming that tools will just catch up doesn't help anyone here that is still stuck with v1 and want to use v2 or v3, for whatever reason.

Screen Shot 2022-01-17 at 5 14 16 PM


Nonetheless, thanks for the code!

**Pull request closed without merging**